### ISSUE TITLE
Cannot import very large fbx: Too many small allocations

### ISSUE THREAD

**Issue by FlorianBorn71** [assimp/assimp#4476]
**Describe the bug**
We have a ~15GB fbx that does not load in AssImp, because the (binary) tokenization and parsing performs more than a billion(!) single allocations (through defines new_Scope, new_Token, which translate to just "new xyz"). The allocator just gives up at some point and throws.
It's confidential customer data, so unfortunately, I cannot share the model for reference.

We have a local workaround where we replace single allocations by block-based allocations and do the cleanup when the model is loaded.
We can clean up that code and keep it in our own fork, but I wonder if you would be interested into getting this solution as a feature into the official version. Note that with such a change it will significantly improve loading time and memory footprint of other large models.

Do you have a suggestion how to approach this?

**Comment by kimkulling** [comment]
Oh my gosh, one billion allocations are much too much. A better allocation scheme would be the solution I guess.

**Comment by FlorianBorn71** [comment]
Do you have any design preference? Or shall we come up with something and you decide whether you want to use it?

**Comment by kimkulling** [comment]
My idea: I would offer a block-based allocator which can be used for the FBX-parser.

What do you think?

**Comment by FlorianBorn71** [comment]
Hi Kim, sounds good!
Do you already have a block allocator class that I should reuse? Alternatively, I can clean up the one I used for the workaround.
Florian

**Comment by kimkulling** [comment]
Indeed, I have one. I have already used this one in another project which had a similar problem with a lot of allocations: https://github.com/kimkulling/cppcore/blob/master/include/cppcore/Memory/TPoolAllocator.h

**Comment by tellypresence** [comment]
Any updates on this?

**Comment by tellypresence** [comment]
FBX import issues tracked via epic #6156
